### PR TITLE
Javascript javascript2 week1

### PR DESCRIPTION
**Pull request by andrii1**

### CODE REVIEW

**Review by nefa10 — COMMENTED** [review]
Great work! 
I would just recommend checking whether the array is empty first, in the findShortestWord function. Although in the exercise the array is given, it is usually a good practice to implement these safety nets, as sometimes they may be called and return something you don't except and ruin the entire code. 
I would actually keep the for loop instead of reduce function because it keeps the readability of the code, but if it works for you then it's fine! 
Overall great work on this one!

File: javascript/javascript2/week1/homework.js [hunk 1]
```diff
@@ -18,3 +20,39 @@ function findShortestWord(words) {
...
+
+function findDanishLetters(string) {
+  const allLettersArray = string.split("");
+  let count1 = 0;
+  let count2 = 0;
+  let count3 = 0;
+  let total = 0;
+  for (let i = 0; i < allLettersArray.length; i++) {
+    for (let j = 0; j < danishCharacters.length; j++) {
+      if (allLettersArray[i] === danishCharacters[j]) {
```
Comment (nefa10) [inline javascript/javascript2/week1/homework.js]: The idea of the two for loops is good, but it is not an also a bit heavy for this. You can try and find some of the implemented JavaScript for string search and also you can try and find other ways of doing this! but overall it works, so good job!

Comment (andrii1) [reply javascript/javascript2/week1/homework.js]: ok, changed to this
`const danishString = "Jeg har en blå bil";
const danishString2 = "Blå grød med røde bær";
const danishCharacters = ["æ", "ø", "å"];

function findDanishLetters(string) {
  const letterAe = danishCharacters[0];
  const letterO = danishCharacters[1];
  const letterAo = danishCharacters[2];
  const countLetterAe = string.split(letterAe).length - 1;
  const countLetterO = string.split(letterO).length - 1;
  const countLetterAo = string.split(letterAo).length - 1;
  let logLetterAe;
  let logLetterO;
  let logLetterAo;

  if (countLetterAe != 0) {
    logLetterAe = `æ: ${countLetterAe}, `;
  } else {
    logLetterAe = "";
  }

  if (countLetterO != 0) {
    logLetterO = `ø: ${countLetterO}, `;
  } else {
    logLetterO = "";
  }

  if (countLetterAo != 0) {
    logLetterAo = `å: ${countLetterAo}`;
  } else {
    logLetterAo = "";
  }

  console.log(
    `{total: ${
      countLetterAe + countLetterO + countLetterAo
    }, ${logLetterAe}${logLetterO}${logLetterAo}}`
  );
}
findDanishLetters(danishString); // returns {total: 1, å: 1}

findDanishLetters(danishString2); // returns {total: 4, æ: 1, ø: 2, å: 1}
`

**Comment by andrii1** [comment]
Also, added a check if array is empty in the findShortestWord function. For some reason, I can't comment below that comment.

**Comment by dpfernandes** [comment]
@27ManishaPatel can you add your reviews on @andrii1 PRs?

File: javascript/javascript2/week1/homework.js (hunk 1 above)
Comment (27ManishaPatel) [reply javascript/javascript2/week1/homework.js]: good job!

**Pull request merged**